src/extension.ts

```diff
@@ -199,6 +200,23 @@ export async function activate(context: vscode.ExtensionContext, perfStats: { lo
     ext.managedIdentityBranchDataProvider = new ManagedIdentityBranchDataProvider();
     ext.v2.api.resources.registerAzureResourceBranchDataProvider(AzExtResourceType.ManagedIdentityUserAssignedIdentities, ext.managedIdentityBranchDataProvider);
 
+
+
+    // Register the exportAuthRecord callback for session changes
+    registerEvent(
+        'treeView.onDidChangeSessions',
+        vscode.authentication.onDidChangeSessions,
+        exportAuthRecord
+    );
+
+    // Also call exportAuthRecord once on activation to ensure the auth record is exported at least once
+    // (onDidChangeSessions is not guaranteed to fire on activation)
+    void exportAuthRecord({
+        errorHandling: {},
+        telemetry: { suppressIfSuccessful: false, properties: {} }
+    } as any);
```

Comment:
I'd rather this be factored out into a second function in `exportAuthRecord.ts`, so that only one line needs to be added to `extension.ts`

Reply:
refactored.